providers/tests/system/google/cloud/cloud_sql/example_cloud_sql_query_iam.py

```diff
@@ -0,0 +1,491 @@
+        ],
+    },
+}
+
+
+def ip_configuration() -> dict[str, Any]:
+    """Generates an ip configuration for a CloudSQL instance creation body"""
+    if run_in_composer():
+        # Use connection to Cloud SQL instance via Private IP within the Cloud Composer's network.
+        return {
+            "ipv4Enabled": True,
+            "requireSsl": False,
+            "sslMode": "ENCRYPTED_ONLY",
+            "enablePrivatePathForGoogleCloudServices": True,
+            "privateNetwork": f"projects/{PROJECT_ID}/global/networks/default",
+        }
+    else:
+        # Use connection to Cloud SQL instance via Public IP from anywhere (mask 0.0.0.0/0).
+        # Consider specifying your network mask
+        # for allowing requests only from the trusted sources, not from anywhere.
+        return {
+            "ipv4Enabled": True,
+            "requireSsl": True,
+            "sslMode": "TRUSTED_CLIENT_CERTIFICATE_REQUIRED",
+            "authorizedNetworks": [
+                {"value": "0.0.0.0/0"},
+            ],
+        }
+
+
+def cloud_sql_instance_create_body(database_provider: dict[str, Any]) -> dict[str, Any]:
+    """Generates a CloudSQL instance creation body"""
+    create_body: dict[str, Any] = deepcopy(CLOUD_SQL_INSTANCE_CREATE_BODY_TEMPLATE)
+    create_body["name"] = database_provider["cloud_sql_instance_name"]
+    create_body["databaseVersion"] = database_provider["database_version"]
+    create_body["settings"]["ipConfiguration"] = ip_configuration()
+    create_body["settings"]["databaseFlags"] = [
+        {"name": database_provider["database_iam_flag_name"], "value": "on"}
+    ]
+    return create_body
+
+
+CLOUD_SQL_DATABASE_NAME = "test_db"
+CLOUD_SQL_USER = "test_user"
+CLOUD_IAM_SA = os.environ.get("CLOUD_IAM_SA", "test_iam_sa")
+CLOUD_SQL_PASSWORD = "JoxHlwrPzwch0gz9"
+CLOUD_SQL_IP_ADDRESS = "127.0.0.1"
+CLOUD_SQL_PUBLIC_PORT = 5432
+
+DB_PROVIDERS: Iterable[dict[str, str]] = (
+    {
+        "database_type": "postgres",
+        "port": "5432",
+        "database_version": "POSTGRES_15",
+        "cloud_sql_instance_name": f"{CLOUD_SQL_INSTANCE_NAME_TEMPLATE}-postgres",
+        "database_iam_flag_name": "cloudsql.iam_authentication",
+        "cloud_sql_iam_sa": CLOUD_IAM_SA.split(".gserviceaccount.com")[0],
+    },
+    {
+        "database_type": "mysql",
+        "port": "3306",
+        "database_version": "MYSQL_8_0",
+        "cloud_sql_instance_name": f"{CLOUD_SQL_INSTANCE_NAME_TEMPLATE}-mysql",
+        "database_iam_flag_name": "cloudsql_iam_authentication",
+        "cloud_sql_iam_sa": CLOUD_IAM_SA,
+    },
+)
+
+
+def cloud_sql_database_create_body(instance: str) -> dict[str, Any]:
+    """Generates a CloudSQL database creation body"""
+    return {
+        "instance": instance,
+        "name": CLOUD_SQL_DATABASE_NAME,
+        "project": PROJECT_ID,
+    }
+
+
+CLOUD_SQL_INSTANCE_NAME = ""
+DATABASE_TYPE = ""  # "postgres|mysql|mssql"
+
+# [START howto_operator_cloudsql_iam_connections]
```

Comment:
Looks like this example has never been used on .rst doc and ci is not happy about it :-(

Reply:
I will add a documentation with the next commit